src/scripts/main.js

```diff
@@ -1,11 +1,19 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve) => {
+    element.addEventListener(eventName, () => {
+      resolve(
+        `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}`,
+      );
+    });
```

Comment:
This usage of `addEventListener` in a `Promise` might lead to unexpected behavior. It's important to remember that the `Promise` will only resolve once, but the event might fire multiple times. This could lead to some events being missed. It would be better to use an event listener outside a `Promise` and handle multiple event firings appropriately.